spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastClientAutoConfiguration.java

```diff
@@ -0,0 +1,99 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.hazelcast;
+
+import java.io.IOException;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastInstance;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnSingleCandidate;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Hazelcast client. Creates a
+ * client {@link HazelcastInstance} based on explicit configuration or when a default
+ * configuration file is found in the environment.
+ *
+ * @author Vedran Pavic
+ * @since 2.0.0
+ * @see HazelcastClientConfigResourceCondition
+ */
+@Configuration
+@ConditionalOnClass(HazelcastClient.class)
+@ConditionalOnMissingBean(HazelcastInstance.class)
+@EnableConfigurationProperties(HazelcastProperties.class)
+public class HazelcastClientAutoConfiguration {
+
+	@Configuration
+	@ConditionalOnMissingBean(ClientConfig.class)
+	@Conditional(ClientConfigAvailableCondition.class)
+	static class HazelcastClientConfigFileConfiguration {
+
+		private final HazelcastProperties hazelcastProperties;
+
+		HazelcastClientConfigFileConfiguration(HazelcastProperties hazelcastProperties) {
+			this.hazelcastProperties = hazelcastProperties;
+		}
+
+		@Bean
+		public HazelcastInstance hazelcastClient() throws IOException {
+			Resource clientConfig = this.hazelcastProperties
+					.resolveClientConfigLocation();
+			if (clientConfig != null) {
+				return new HazelcastClientInstanceFactory(clientConfig)
+						.getHazelcastInstance();
+			}
+			return HazelcastClient.newHazelcastClient();
+		}
+
+	}
+
+	@Configuration
+	@ConditionalOnSingleCandidate(ClientConfig.class)
+	static class HazelcastClientConfigConfiguration {
+
+		@Bean
+		public HazelcastInstance hazelcastClient(ClientConfig clientConfig) {
+			return new HazelcastClientInstanceFactory(clientConfig)
+					.getHazelcastInstance();
+		}
+
+	}
+
+	/**
+	 * {@link HazelcastClientConfigResourceCondition} that checks if the
+	 * {@code spring.hazelcast.client-config} configuration key is defined.
+	 */
+	static class ClientConfigAvailableCondition
+			extends HazelcastClientConfigResourceCondition {
+
+		ClientConfigAvailableCondition() {
+			super("spring.hazelcast", "client-config");
```

Comment:
Is there a reason why you'd want to have a dedicated property for this. If you look at how Hazelcast has implemented the JSR-107 support they have a server and client flag (that leads to two `CacheProvider` implementations and each of them read the same properties as far as I know). I would not deviate from that so, IMO, that should read `spring.hazelcast.config`.